dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -417,9 +422,19 @@ private LambdaEvent createBasicEvent(String repository, String gitReference, Str
 
                 String subclass = wf.getSubclass();
                 String workflowName = wf.getName();
+                Boolean publish = wf.getPublish();
 
                 Workflow workflow = createOrGetWorkflow(BioWorkflow.class, repository, user, workflowName, subclass, gitHubSourceCodeRepo);
                 workflow = addDockstoreYmlVersionToWorkflow(repository, gitReference, dockstoreYml, gitHubSourceCodeRepo, workflow);
+
+                if (publish != null) {
+                    try {
+                        workflow = publishWorkflow(workflow, publish);
+                    } catch (CustomWebApplicationException ex) {
+                        LOG.warn("Could not set publish state from YML.", ex.getMessage());
```

Comment:
Log the exception proper as well.

Hmm, the user will have no way of knowing this failed. Can/should we add it to GitHub app logs? I realize it will probably be just as easy to just see if something is published or not, but if it's in the logs, it gives us the option of doing something in the future.

Reply:
(Added the full exception to the log)

Yeah, I was thinking that it should be easy enough to see if it got published (it's the same or less effort than looking in GH App logs).
Furthermore, if the publish state isn't as the user expects, they can try manually publishing like normal to see any error.

Anyhow if you or @denis-yuen think it's important, I can look into getting this into GH Apps logs.

Reply:
Yes, let's say get this if easy (a couple hours work), new ticket it more involved than that.

Reply:
Added a `PUBLISH` LambdaEvent to indicate success/failure of YML publish/unpublish operations.